lib/portage/repository/config.py

```diff
@@ -338,13 +342,20 @@ def __init__(self, name, repo_opts, local_config=True):
 				self.name = layout_data['repo-name']
 				self.missing_repo_name = False
 
+			# Override some repos.conf setting with layout.conf
 			for value in ('allow-missing-manifest',
 				'cache-formats',
-				'create-manifest', 'disable-manifest', 'manifest-hashes',
+				'create-manifest', 'disable-manifest', 'manifest-hashes', 'maintainer-nag', 'maintainer-nag-url',
 				'manifest-required-hashes', 'profile-formats', 'properties-allowed', 'restrict-allowed',
 				'sign-commit', 'sign-manifest', 'thin-manifest', 'update-changelog'):
 				setattr(self, value.lower().replace("-", "_"), layout_data[value])
 
+			# Allow repos.conf override of layout.conf for these
+			for value in ('maintainer-nag', 'maintainer-nag-url'):
+				setting = repo_opts.get(value)
+				if setting:
+					setattr(self, value.lower().replace("-", "_"), setting)
+
```

Comment:
When testing, I needed to add something like this to ensure that `maintainer-nag = no` translated to a boolean False, otherwise I was unable to disable the warning:
```python
			self.maintainer_nag = (self.maintainer_nag.lower() in ('true', 'yes')
				if isinstance(self.maintainer_nag, basestring) else self.maintainer_nag)
```